packages/hierarchies/learning/PresentationRulesMigrationGuide.md

```diff
@@ -0,0 +1,779 @@
+        grouping: {
+          byProperties: {
+            propertiesClassName: "BisCore.GeometricElement3d",
+            createGroupForOutOfRangeValues: true,
+            createGroupForUnspecifiedValues: true,
+            propertyGroups: [
+              {
+                propertyClassAlias: "this",
+                propertyName: "Yaw",
+                ranges: [
+                  { fromValue: 0, toValue: 0, rangeLabel: "Zero" },
+                  { fromValue: -360, toValue: 0, rangeLabel: "Negative" },
+                  { fromValue: 0, toValue: 360, rangeLabel: "Positive" },
+                ],
+              },
+            ],
+          },
+        },
+      })}
+      FROM BisCore.GeometricElement3d [this]
+    `,
+  },
+};
+```
+
+<!-- END EXTRACTION -->
+
+### Migrating label grouping
+
+Label grouping option, simply puts instance nodes under label grouping nodes based on node's label. In Presentation Rules this grouping is set at hierarchy specification level, similar to [class grouping](#migrating-class-grouping):
+
+```json
+{
+  "specType": "InstanceNodesOfSpecificClasses",
+  "classes": { "schemaName": "BisCore", "classNames": ["Element"], "arePolymorphic": true },
+  "groupByLabel": true
+}
+```
+
+Matching `HierarchyDefinition`:
+
+<!-- [[include: [Presentation.Hierarchies.Migration.HierarchyNodesDefinitionImports, Presentation.Hierarchies.Migration.LabelGrouping], ts]] -->
+<!-- BEGIN EXTRACTION -->
+
+```ts
+import { createNodesQueryClauseFactory, HierarchyLevelDefinition, HierarchyNodesDefinition, InstancesNodeKey } from "@itwin/presentation-hierarchies";
+import { createBisInstanceLabelSelectClauseFactory } from "@itwin/presentation-shared";
+
+const selectClauseFactory = createNodesQueryClauseFactory({ imodelAccess });
+const labelsFactory = createBisInstanceLabelSelectClauseFactory({ classHierarchyInspector: imodelAccess });
+const definition: HierarchyNodesDefinition = {
+  fullClassName: "BisCore.Element",
+  query: {
+    ecsql: `
+      SELECT ${await selectClauseFactory.createSelectClause({
+        ecClassId: { selector: "this.ECClassId" },
+        ecInstanceId: { selector: "this.ECInstanceId" },
+        nodeLabel: { selector: await labelsFactory.createSelectClause({ className: "BisCore.Element", classAlias: "this" }) },
+        grouping: {
+          byLabel: { hideIfNoSiblings: true, hideIfOneGroupedNode: true },
+        },
+      })}
+      FROM BisCore.Element [this]
+    `,
+  },
+};
+```
+
+<!-- END EXTRACTION -->
+
+### Migrating same label grouping
+
+Same label grouping is different from other types of groupings, because it doesn't result in a grouping node being created. Instead, nodes with the same label get merged into a single instance node, which represents a merged set of instances. In the Presentation Rules system there are two options to choose from - grouping at query stage or at post-processing stage. The first one is more performance while the second one is more flexible.
```

Comment:
```suggestion
Same label grouping is different from other types of groupings, because it doesn't result in a grouping node being created. Instead, nodes with the same label get merged into a single instance node, which represents a merged set of instances. In the Presentation Rules system there are two options to choose from - grouping at query stage or at post-processing stage. The first one is more performant while the second one is more flexible.
```